frontend/src/components/CalcSpectrum.tsx

```diff
@@ -163,7 +166,40 @@ export const CalcSpectrum: React.FC = () => {
           setCalcSpectrumResponse(response);
         }
       }
-      setLoading(false); //setLoading(false) is called after the response is received
+      setLoading(false);
+      setdownloadbutton(false);
+    });
+  };
+  //download the .spec file
+  const downloadSpec = async (data: FormValues): Promise<void> => {
+    if (useSlit == true) {
+      if (data.mode === "radiance_noslit") {
+        data.mode = "radiance";
+      }
+      if (data.mode === "transmittance_noslit") {
+        data.mode = "transmittance";
+      }
+    }
+    console.log(data);
+    setError(undefined);
+    import(/* webpackIgnore: true */ "./config.js").then(async (module) => {
+      const rawResponse = await axios.post(
+        module.apiEndpoint + `download_spectrum`,
```

Comment:
```suggestion
        module.apiEndpoint + `download-spectrum`,
```
The endpoints should be consistent in how they are named (so either use _ for everything or - for everything)